.github/workflows/docs.yml

```diff
@@ -0,0 +1,105 @@
+name: docs
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  check-links:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+
+    - uses: actions/setup-python@v3
+      with:
+        python-version: '3.10'
+
+    - name: Create virtualenv
+      run: python3 -m venv .venv
+
+    - name: Install dependencies
+      run: |
+        source .venv/bin/activate
+        python -m pip install .
+        python -m pip install -r docs/requirements.txt
+
+    - name: Linkcheck
+      working-directory: docs
+      run: |
+        source ../.venv/bin/activate
+
+        set +e
+        make linkcheck
+        exit_code=$?
+
+        set -e
+
+        if [ $exit_code -eq 0 ]; then
+            echo -e "\n\n=================\nAll links are valid!"
+
+            echo "# :heavy_check_mark: Sphinx links" >> $GITHUB_STEP_SUMMARY
```

Comment:
You can see the result of this in https://github.com/JeanChristopheMorinPerso/OpenTimelineIO/actions/runs/2521426263/attempts/1#summary-6950080420 when it succeeds and https://github.com/JeanChristopheMorinPerso/OpenTimelineIO/actions/runs/2521417320/attempts/1#summary-6950061883 when it fails. I thought it would be a nice addition that can be later improved even more.